### ISSUE TITLE
Arearange series with very close x-values fail to render Chart

### ISSUE THREAD

**Issue by nishikantparmariam** [highcharts/highcharts#20057]
#### Expected behaviour
Arearange series should always render.

#### Actual behaviour
Arearange series does not render if x-values are too close. This data was generated using `numpy.arange(1, 12, 0.01)`.

#### Live demo with steps to reproduce
**https://jsfiddle.net/vpjxzg85/1/**

#### Product version
10.3.3

#### Affected browser(s)
Chrome

**Comment by bm64** [comment]
Hello @nishikantparmariam 

It seems the issue stems from the `turboThreshold` property, which activates when there are more than 1000 data points. Below this threshold, the problem doesn't occur: https://jsfiddle.net/BlackLabel/6cph059m/

There's a compatibility problem with `turboThreshold` when using `keys`; removing keys resolves the issue: https://jsfiddle.net/BlackLabel/bfr2n8oa/

As a workaround you disable `turboThreshold` by setting it to 0: https://jsfiddle.net/BlackLabel/n9h3sqau/

**Comment by oysteinmoseng** [comment]
Keys are currently not supported in turbo-mode, but we will discuss how to proceed here. At minimum we should improve docs to reflect this.

**Comment by nishikantparmariam** [comment]
Thanks for the workaround @bm64 @oysteinmoseng! I had another bigger question - 

If series `data` is passed as 2D arrays, then for multi-dim series (like arearange, errorbar, heatmap, boxplot), I see that there is a **standard** order of `keys` denoting what each value is, in a single point's array e.g. if `data` is  `[[a0, b0, c0], [a1, b1, c1], [a2, b2, c2], [a3, b3, c3]])` and series type is say `arearange`, then Highcharts assumes that first value is x, second is low and third is high. 

_As you said that, turbo mode is incompatible with `keys`, so if I do not pass `keys` but ensure that each point's value is in correct order (as per that series type), will I gain **performance improvements** in rendering chart as compared to passing `keys` explicity - the turbo mode?_

**Comment by bm64** [comment]
@nishikantparmariam 

Yes, you're right. In Highcharts, using 2D arrays in the format specified by the API for specific series types, like arearange, enables turbo mode when the data exceeds the turboThreshold, which defaults to 1000 for arearange series. This mode skips extra data checks, enhancing performance for large datasets. Note that turbo and boost modes work only with array configurations and are not compatible with object point configurations.
